examples/window/window_settings.rs

```diff
@@ -124,6 +126,30 @@ fn toggle_window_controls(input: Res<ButtonInput<KeyCode>>, mut windows: Query<&
     }
 }
 
+fn toggle_resolutions(input: Res<ButtonInput<KeyCode>>, mut windows: Query<&mut Window>) {
+    if input.just_pressed(KeyCode::KeyR) {
+        let mut window = windows.single_mut();
+        let current_resolution = CommonScreenResolution::iter()
+            .find(|&r| WindowResolution::from(r) == window.resolution);
+        window.resolution = if current_resolution.is_none() {
+            CommonScreenResolution::R360p.into()
+        } else {
+            let resolutions = CommonScreenResolution::iter()
+                .map(std::convert::Into::into)
+                .collect::<Vec<CommonScreenResolution>>();
+            let current_resolution_index = CommonScreenResolution::iter()
+                .position(|r| r == current_resolution.unwrap())
+                .unwrap();
+
+            if current_resolution_index + 1 >= resolutions.len() {
+                resolutions[0].into()
+            } else {
+                resolutions[current_resolution_index + 1].into()
+            }
+        }
+    }
+}
```

Comment:
This would be more idiomatic if `CurrentResolution` was a `Resource` initialized at `CommonScreenResolution::R360p` imo

Reply:
You can use `Iterator::cycle()`, store that as a resource and call `next` every time input is pressed